<?xml version="1.0" ?>
<robot name="New" xmlns:xacro="http://www.ros.org/wiki/xacro">

<xacro:include filename="$(find New_description)/urdf/materials.xacro" />
<xacro:include filename="$(find New_description)/urdf/New.trans" />
<xacro:include filename="$(find New_description)/urdf/New.gazebo" />
<link name="base_link">
  <inertial>
    <origin xyz="-0.04149471418155845 -0.0025971159520870152 0.013779631134822855" rpy="0 0 0"/>
    <mass value="0.5842942923229515"/>
    <inertia ixx="0.00031" iyy="0.001251" izz="0.001356" ixy="-7.7e-05" iyz="6e-06" ixz="2.3e-05"/>
  </inertial>
  <visual>
    <origin xyz="0 0 0" rpy="0 0 0"/>
    <geometry>
      <mesh filename="package://New_description/meshes/base_link.stl" scale="0.001 0.001 0.001"/>
    </geometry>
    <material name="silver"/>
  </visual>
  <collision>
    <origin xyz="0 0 0" rpy="0 0 0"/>
    <geometry>
      <mesh filename="package://New_description/meshes/base_link.stl" scale="0.001 0.001 0.001"/>
    </geometry>
  </collision>
</link>

<link name="link_1_1">
  <inertial>
    <origin xyz="0.001392191052634139 -0.009015999327819468 0.0235284962411932" rpy="0 0 0"/>
    <mass value="0.2524132842018846"/>
    <inertia ixx="0.000128" iyy="0.000137" izz="7.6e-05" ixy="2e-06" iyz="1.5e-05" ixz="8e-06"/>
  </inertial>
  <visual>
    <origin xyz="-0.000476 -0.0 -0.033399" rpy="0 0 0"/>
    <geometry>
      <mesh filename="package://New_description/meshes/link_1_1.stl" scale="0.001 0.001 0.001"/>
    </geometry>
    <material name="silver"/>
  </visual>
  <collision>
    <origin xyz="-0.000476 -0.0 -0.033399" rpy="0 0 0"/>
    <geometry>
      <mesh filename="package://New_description/meshes/link_1_1.stl" scale="0.001 0.001 0.001"/>
    </geometry>
  </collision>
</link>

<link name="link_2_1">
  <inertial>
    <origin xyz="0.00014024094001882127 0.007326916010056965 0.030962792328760272" rpy="0 0 0"/>
    <mass value="0.13407278823863725"/>
    <inertia ixx="0.000109" iyy="0.000112" izz="1e-05" ixy="-0.0" iyz="1e-06" ixz="-4e-06"/>
  </inertial>
  <visual>
    <origin xyz="0.009476 0.001031 -0.085858" rpy="0 0 0"/>
    <geometry>
      <mesh filename="package://New_description/meshes/link_2_1.stl" scale="0.001 0.001 0.001"/>
    </geometry>
    <material name="silver"/>
  </visual>
  <collision>
    <origin xyz="0.009476 0.001031 -0.085858" rpy="0 0 0"/>
    <geometry>
      <mesh filename="package://New_description/meshes/link_2_1.stl" scale="0.001 0.001 0.001"/>
    </geometry>
  </collision>
</link>

<link name="link_3_1">
  <inertial>
    <origin xyz="2.7826710637217367e-06 -0.006259371756109719 0.026211019639232058" rpy="0 0 0"/>
    <mass value="0.10702282683217142"/>
    <inertia ixx="4.7e-05" iyy="5.1e-05" izz="8e-06" ixy="0.0" iyz="0.0" ixz="0.0"/>
  </inertial>
  <visual>
    <origin xyz="0.006834 0.001231 -0.148989" rpy="0 0 0"/>
    <geometry>
      <mesh filename="package://New_description/meshes/link_3_1.stl" scale="0.001 0.001 0.001"/>
    </geometry>
    <material name="silver"/>
  </visual>
  <collision>
    <origin xyz="0.006834 0.001231 -0.148989" rpy="0 0 0"/>
    <geometry>
      <mesh filename="package://New_description/meshes/link_3_1.stl" scale="0.001 0.001 0.001"/>
    </geometry>
  </collision>
</link>

<link name="link_4_1">
  <inertial>
    <origin xyz="0.00016784198048194927 0.016597930425015908 0.0023250889260546803" rpy="0 0 0"/>
    <mass value="0.07823910234886815"/>
    <inertia ixx="1.2e-05" iyy="1e-05" izz="9e-06" ixy="-0.0" iyz="-0.0" ixz="0.0"/>
  </inertial>
  <visual>
    <origin xyz="0.006957 0.001301 -0.201505" rpy="0 0 0"/>
    <geometry>
      <mesh filename="package://New_description/meshes/link_4_1.stl" scale="0.001 0.001 0.001"/>
    </geometry>
    <material name="silver"/>
  </visual>
  <collision>
    <origin xyz="0.006957 0.001301 -0.201505" rpy="0 0 0"/>
    <geometry>
      <mesh filename="package://New_description/meshes/link_4_1.stl" scale="0.001 0.001 0.001"/>
    </geometry>
  </collision>
</link>

<joint name="Joint_1" type="continuous">
  <origin xyz="0.000476 0.0 0.033399" rpy="0 0 0"/>
  <parent link="base_link"/>
  <child link="link_1_1"/>
  <axis xyz="0.0 0.0 1.0"/>
</joint>

<joint name="Joint_2" type="continuous">
  <origin xyz="-0.009952 -0.001031 0.052459" rpy="0 0 0"/>
  <parent link="link_1_1"/>
  <child link="link_2_1"/>
  <axis xyz="0.0 1.0 0.0"/>
</joint>

<joint name="Joint_3" type="continuous">
  <origin xyz="0.002642 -0.0002 0.063131" rpy="0 0 0"/>
  <parent link="link_2_1"/>
  <child link="link_3_1"/>
  <axis xyz="0.0 -1.0 0.0"/>
</joint>

<joint name="Joint_4" type="continuous">
  <origin xyz="-0.000123 -7e-05 0.052516" rpy="0 0 0"/>
  <parent link="link_3_1"/>
  <child link="link_4_1"/>
  <axis xyz="0.0 1.0 0.0"/>
</joint>

<!-- Transmissions for ros_control -->
<transmission name="Joint_1_trans">
  <type>transmission_interface/SimpleTransmission</type>
  <joint name="Joint_1">
    <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
  </joint>
  <actuator name="Joint_1_motor">
    <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    <mechanicalReduction>1</mechanicalReduction>
  </actuator>
</transmission>

<transmission name="Joint_2_trans">
  <type>transmission_interface/SimpleTransmission</type>
  <joint name="Joint_2">
    <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
  </joint>
  <actuator name="Joint_2_motor">
    <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    <mechanicalReduction>1</mechanicalReduction>
  </actuator>
</transmission>

<transmission name="Joint_3_trans">
  <type>transmission_interface/SimpleTransmission</type>
  <joint name="Joint_3">
    <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
  </joint>
  <actuator name="Joint_3_motor">
    <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    <mechanicalReduction>1</mechanicalReduction>
  </actuator>
</transmission>

<transmission name="Joint_4_trans">
  <type>transmission_interface/SimpleTransmission</type>
  <joint name="Joint_4">
    <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
  </joint>
  <actuator name="Joint_4_motor">
    <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    <mechanicalReduction>1</mechanicalReduction>
  </actuator>
</transmission>

</robot>
